sbt-plugin/src/main/scala/scala/scalajs/sbtplugin/ScalaJSPluginInternal.scala

```diff
@@ -632,6 +640,25 @@ object ScalaJSPluginInternal {
     }
   }
 
+  protected def javaSysProps2JSLib(
+      javaOptions: Seq[(String, String)]): Option[ResolvedJSDependency] = {
```

Comment:
This should still be in its own task to modularize its constructions wrt `loadedJSEnv`. I suggest `scalaJSConfigurationLibs: TaskKey[Seq[ResolvedJSDependency]]`, initialized to `Nil` in the `projectBaseSettings` (or even in `Global`?), and in `configSettings` we would `++=` the content of this method.

That will allow users to inject their own configuration files very easily, if required.